### ISSUE TITLE
nnf-controler-manager uses UTC timestamps

### ISSUE THREAD

**Issue by behlendorf** [NearNodeFlash/NearNodeFlash.github.io#36]
When viewing the `nnf-controller-manager` container logs they use a UTC based timestamp. e.g.  2023-02-27T19:06:31.751Z (note the trailing Z)

```
2023-02-27T19:06:31.751Z    INFO    controllers.NnfWorkflow    Start    {"Workflow": "default/fluxjob-187278588523840512", "state": "PreRun", "index": 0, "directive": "#DW jobdw capacity=1TiB type=xfs name=test1"}                                                   │
2023-02-27T19:06:31.751Z    INFO    controllers.NnfWorkflow    Start done    {"Workflow": "default/fluxjob-187278588523840512", "state": "PreRun", "index": 0}   
```

Other logs like those from the `nnf-nodemanager` use the local time.
  
  ```
2023-02-27T10:35:58.314-0800    INFO    controllers.NnfNodeStorage    Created filesystem    {"NnfNodeStorage": "hetchy44/default-fluxjob-187285641715778560-0-xfs-0", "Id": "default-fluxjob-187285641715778560-0-xfs-0-0"}                                             │
  ```
  
Can the `nnf-controller-manager` be configured or updated to use local time to make it easier to correlate events between log files.

**Comment by ajfloeder** [comment]
@behlendorf are all of your systems (k8s-master and the Rabbits) configured to use the same time zone and NTP settings?

**Comment by behlendorf** [comment]
Yes, the configuration is the same on all the nodes running k8s:

```
# pdsh -w hetchy[4-6,9,44] timedatectl | dshbak -c
----------------
hetchy[4-6,9,44]
----------------
               Local time: Wed 2023-03-01 13:34:12 PST
           Universal time: Wed 2023-03-01 21:34:12 UTC
                 RTC time: Wed 2023-03-01 21:34:12
                Time zone: US/Pacific (PST, -0800)
System clock synchronized: yes
              NTP service: n/a
          RTC in local TZ: no
```

**Comment by bdevcich** [comment]
@behlendorf could you also confirm that those timezones are linked to `/usr/share/zoneinfo/...`? 

Example:
```
clush -w 'rabbit-k8s-master,rabbit-k8s-worker,rabbit-node-[1,2]' -b 'ls -l /etc/localtime'
---------------
rabbit-k8s-master
---------------
lrwxrwxrwx. 1 root root 37 Nov 28  2022 /etc/localtime -> ../usr/share/zoneinfo/America/Chicago
---------------
rabbit-k8s-worker
---------------
lrwxrwxrwx. 1 root root 37 Oct 14 12:45 /etc/localtime -> ../usr/share/zoneinfo/America/Chicago
---------------
rabbit-node-1
---------------
lrwxrwxrwx. 1 root root 37 May 17  2022 /etc/localtime -> ../usr/share/zoneinfo/America/Chicago
---------------
rabbit-node-2
---------------
lrwxrwxrwx. 1 root root 37 May 12  2022 /etc/localtime -> ../usr/share/zoneinfo/America/Chicago
```

We've determined that this is most likely due to the node manager pods mounting in the hosts' `/etc` directory, which means we get `/etc/localtime` as a side effect. Only the node manager pods are doing this, which is why those are in local time since the rabbit nodes are using local time. And why the node-controller-manager are not inheriting this from the k8s-workers nodes. By default, pods are in UTC time.

I believe we would want to add mounts into non node-manager pods for both `/etc/localtime` and `/usr/share/zoneinfo` to get all of the NNF pods to use local time instead of defaulting to UTC.

**Comment by behlendorf** [comment]
@bdevcich that looks to be the case.  On all of the k8s master nodes, k8s worker nodes, and rabbit nodes `/etc/localtime` is linked to `/usr/share/zoneinfo/US/Pacific`.  Adding these mounts to get them all using local time make sense to me.

**Comment by bdevcich** [comment]
It looks like this approach will need both belt and suspenders. For example, the nnf-dm pods do not respond to mounting in `/etc/localtime`. I have not yet determined why, but the symlink in those pods stay stuck to UTC.

```
root@8e013f15edaf:/# ls -l /etc/localtime
lrwxrwxrwx 1 root root 27 Sep 26 00:00 /etc/localtime -> /usr/share/zoneinfo/Etc/UTC
```

However, I did try setting the `TZ` environment variable and that did work for the nnf-dm pods even with the `/etc/localtime` symlink to UTC. Using this variable will require a timezone to be set explicitly or we need to use InitContainers to do this programmatically from the existing `/etc/localtime` symlink on the host.

Then will then need to be applied to every NNF pod across all repositories to remain consistent.

**Comment by bdevcich** [comment]
This work as been completed across all NNF pods and containers with the exception of the `kube-rbac-proxy` containers. All containers will inherit the host node's `/etc/localtime` and `/usr/share/zoneinfo`.
